tests/net/nanocoap_cli/nanocli_client.c

```diff
@@ -164,9 +172,69 @@ static int _cmd_client(int argc, char **argv)
            argv[0]);
     return 1;
 }
-
 SHELL_COMMAND(client, "CoAP client", _cmd_client);
 
+static bool _get_nibble(uint8_t *dest, char _c)
+{
+    uint8_t c = _c;
+    if (((uint8_t)'0' <= c) && (c <= (uint8_t)'9')) {
+        *dest = c - (uint8_t)'0';
+        return true;
+    }
+
+    if (((uint8_t)'a' <= c) && (c <= (uint8_t)'f')) {
+        *dest = c - (uint8_t)'a' + 10;
+        return true;
+    }
+
+    if (((uint8_t)'A' <= c) && (c <= (uint8_t)'F')) {
+        *dest = c - (uint8_t)'A' + 10;
+        return true;
+    }
+
+    return false;
+}
+
+static int _cmd_client_token(int argc, char **argv){
+    if (argc != 2) {
+        printf("Usage: %s <TOKEN_HEX>\n", argv[0]);
+        return 1;
+    }
+
+    size_t len = strlen(argv[1]);
+    if (len & 1) {
+        puts("Token invalid: Must be an even number of hex chars");
+        return 1;
+    }
+
+    if (len > 2 * sizeof(_client_token)) {
+        printf("Token too long (%u B, but max is %u B)\n",
+               (unsigned)(len / 2), (unsigned)sizeof(_client_token));
+        return 1;
+    }
+
+    _client_token_len = len / 2;
+
+    for (size_t pos = 0; pos < _client_token_len; pos++) {
+        uint8_t high, low;
+        if (!_get_nibble(&high, argv[1][pos << 1])) {
+            puts("Token invalid: Must be given in hex");
+            return 1;
+        }
+
+        if (!_get_nibble(&low, argv[1][(pos << 1) + 1])) {
+            puts("Token invalid: Must be given in hex");
+            return 1;
+        }
+
+        _client_token[pos] = (high << 4) | low;
```

Comment:
Maybe move that to `scn_buf_hex(const char *str, size_t str_len, uint8_t *out, size_t out_len)`

Reply:
Done